### PR TITLE
Update `switch_face` to `attachment_face` and add bells

### PR DESCRIPTION
**Pull request by BreadcrumbIsTaken**
## ~~`attachment` property for bells~~ ~~add bells to `switch_face`~~ `attachment_face`

Changes the the attachment type of bells.

Requested by [Gyro](https://discord.com/channels/315163488085475337/1133153349257154621)

### CODE REVIEW

**Comment by BreadcrumbIsTaken** [comment]
Done

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java [hunk 1]
```diff
@@ -3,99 +3,94 @@
...
-            case FLOOR:
-                return BlockFace.DOWN;
-            case CEILING:
-                return BlockFace.UP;
-            default:
-                return BlockFace.SELF;
+                case FLOOR -> BlockFace.DOWN;
+                case CEILING -> BlockFace.UP;
+            };
         }
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java]: should support bells here?

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java [hunk 2]
```diff
@@ -3,99 +3,94 @@
...
-public class MaterialSwitchFace implements Property {
+public class MaterialSwitchFace extends MaterialProperty<ElementTag> {
 
-    public static boolean describes(ObjectTag material) {
-        return material instanceof MaterialTag
-                && ((MaterialTag) material).hasModernData()
-                && ((MaterialTag) material).getModernData() instanceof FaceAttachable;
+    // <--[property]
+    // @object MaterialTag
+    // @name switch_face
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java]: probably time to rename this property to be more generic, given that it's very much not switch-specific

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java [hunk 3]
```diff
@@ -3,99 +3,94 @@
...
+            return new MaterialSwitchFace(_material);
         }
     }
 
-    public static final String[] handledMechs = new String[] {
-            "switch_face"
-    };
-
     public MaterialSwitchFace(MaterialTag _material) {
         material = _material;
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java]: this constructor and field should be discarded with the format update

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java [hunk 4]
```diff
@@ -3,99 +3,94 @@
...
-        // @input ElementTag
-        // @description
-        // Sets the current attach direction for a switch or other attachable material.
-        // @tags
-        // <MaterialTag.switch_face>
-        // -->
-        if (mechanism.matches("switch_face") && mechanism.requireEnum(FaceAttachable.AttachedFace.class)) {
-            getFaceAttachable().setAttachedFace(FaceAttachable.AttachedFace.valueOf(mechanism.getValue().asString().toUpperCase()));
-        }
+    private BlockData getData() {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchFace.java]: redundant, `getBlockData` exists within `MaterialProperty`

**Comment by BreadcrumbIsTaken** [comment]
Done

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttachmentFace.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                case FLOOR -> BlockFace.DOWN;
+                case CEILING -> BlockFace.UP;
+            };
+        }
+        else if (getBlockData() instanceof Bell bell) {
+            return switch (bell.getAttachment()) {
+                case SINGLE_WALL -> ((Directional) getBlockData()).getFacing();
+                case FLOOR -> BlockFace.DOWN;
+                case CEILING -> BlockFace.UP;
+                default -> null;
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttachmentFace.java]: This default is redundant, also maybe should add double_wall if Directional supports it still

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttachmentFace.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+public class MaterialAttachmentFace extends MaterialProperty<ElementTag> {
+
+    // <--[property]
+    // @object MaterialTag
+    // @name attachment_face
+    // @input ElementTag
+    // @description
+    // Controls the current attach direction for attachable materials such as switches, grindstones, and bells.
+    // Values are "CEILING", "FLOOR", or "WALL". For bells, values are "CEILING", "FLOOR", "SINGLE_WALL", or "DOUBLE_WALL".
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialAttachmentFace.java]:
```
For bells, values are "CEILING", "FLOOR", "SINGLE_WALL", or "DOUBLE_WALL".
For all other supported types, values are "CEILING", "FLOOR", or "WALL".
```

**Comment by BreadcrumbIsTaken** [comment]
Done x3

**Pull request merged**